Scripts/Runtime/Entities/TaskSystem/TaskStream/TaskStream.cs

```diff
@@ -7,30 +9,88 @@ namespace Anvil.Unity.DOTS.Entities.Tasks
     /// <see cref="AbstractTaskSystem"/>.
     /// </summary>
     /// <typeparam name="TInstance">The type of <see cref="IEntityProxyInstance"/> in this stream</typeparam>
+    // ReSharper disable once ClassNeverInstantiated.Global
     public class TaskStream<TInstance> : AbstractTaskStream
         where TInstance : unmanaged, IEntityProxyInstance
     {
+        private readonly string m_TypeString;
+        
         internal EntityProxyDataStream<TInstance> DataStream { get; }
+        internal EntityProxyDataStream<TInstance> PendingCancelDataStream { get; }
+        internal TaskStreamFlags Flags { get; }
 
-        internal override bool IsCancellable
+
+        internal sealed override bool IsCancellable
         {
-            get => false;
+            get => (Flags & TaskStreamFlags.IsCancellable) != 0;
         }
 
-        internal TaskStream()
+        internal sealed override bool IsDataStreamAResolveTarget
+        {
+            get => (Flags & TaskStreamFlags.IsResolveTarget) != 0;
+        }
+        
+        internal TaskStream(TaskStreamFlags flags)
         {
-            //TODO: #71 - This is weird with no dispose but #71 will fix it.
+            Flags = flags;
```

Comment:
Do we need to assert valid flag combinations? Can this be cancellable without being a resolve target?

Reply:
No need currently. All combinations are valid.